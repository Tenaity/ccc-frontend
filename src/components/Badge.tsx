import React from "react";
import type { ShiftCode } from "../types";
import { cn } from "../lib/utils";

type Variant =
    | "td"            // mặc định: trực tại Tổng đài (ngày)
    | "pgd"           // trực PGD (ngày hoặc đêm)
    | "k-white"       // K nền trắng (thứ 7, TD)
    | "leader-day"    // K @ TD (trưởng ca ngày)
    | "leader-night"  // Đ @ TD (trưởng ca đêm)
    | "night-white"   // Đ trắng @ TD
    | "night-pgd";    // Đ @ PGD

export default function Badge({
    code,
    variant = "td",
    crown,
    rank,
    pinned,
}: {
    code: ShiftCode | "";
    variant?: Variant;
    /** legacy prop from old code */
    crown?: boolean;
    pinned?: boolean;
    rank?: number;
}) {
    if (!code) return <span className="text-muted-foreground">—</span>;

    const variantClass: Record<Variant, string> = {
        td: "",
        pgd: "bg-rose-100 border-rose-400",
        "k-white": "bg-white border-dashed border-gray-400",
        "leader-day":
            "border-2 border-green-600 shadow-[0_0_0_1px_rgba(22,163,74,0.12)_inset]",
        "leader-night":
            "border-2 border-violet-600 shadow-[0_0_0_1px_rgba(124,58,237,0.12)_inset]",
        "night-white": "bg-white border-rose-200",
        "night-pgd": "bg-rose-200 border-rose-400",
    };

    const rankClass =
        rank === 1
            ? "border-2 border-gray-900"
            : rank === 2
                ? "border border-dashed border-gray-400"
                : "border";

    // 👑: hiển thị nếu crown=true hoặc là leader
    const showCrown =
        !!crown || variant === "leader-day" || variant === "leader-night";

    return (
        <span
            className={cn(
                "inline-flex items-center gap-1.5 px-1.5 py-0.5 rounded-md text-xs font-medium",
                pickBg(code),
                rankClass,
                variantClass[variant]
            )}
        >
            {pinned ? <span title="Fixed" className="mr-1">📌</span> : null}
            {code}
            {showCrown ? <span title="Trưởng ca" className="ml-1">👑</span> : null}
        </span>
    );
}

function pickBg(code: ShiftCode | "") {
    switch (code) {
        case "CA1":
            return "bg-blue-100";
        case "CA2":
            return "bg-orange-100";
        case "K":
            return "bg-green-100";
        case "HC":
            return "bg-indigo-100";
        case "Đ":
            return "bg-rose-100";
        case "P":
            return "bg-gray-200";
        default:
            return "bg-gray-100";
    }
}