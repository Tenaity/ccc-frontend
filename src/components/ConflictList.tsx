import React from "react";

export default function ConflictList({ conflicts }: { conflicts: any[] }) {
  if (!conflicts || conflicts.length === 0) return null;
  return (
    <div style={{ background: "#FEF2F2", border: "1px solid #FCA5A5", padding: 12, marginBottom: 12, borderRadius: 8 }}>
      <strong>Conflicts:</strong>
      <ul style={{ margin: "8px 0 0 16px" }}>
        {conflicts.map((c, idx) => (
          <li key={idx} style={{ marginBottom: 4 }}>
            {c.day}: {c.type}
          </li>
        ))}
      </ul>
    </div>
  );
}
