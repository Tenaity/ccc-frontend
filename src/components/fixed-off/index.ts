export { default as FixedOffPanel } from './FixedOffPanel';
