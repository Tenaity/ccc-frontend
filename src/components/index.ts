export * from './fixed-off';
export { default as Toolbar } from './Toolbar';
