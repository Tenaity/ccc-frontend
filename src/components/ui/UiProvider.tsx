import * as React from "react"

import { TooltipProvider } from "@/components/ui/tooltip"
import { Toaster } from "@/components/ui/sonner"
import { ToastStateProvider } from "@/components/ui/use-toast"
import { ThemeProvider } from "@/components/ui/theme-provider"

export function UiProvider({ children }: { children: React.ReactNode }) {
  return (
    <ThemeProvider defaultTheme="system" storageKey="customer-care-theme">
      <TooltipProvider delayDuration={200}>
        <ToastStateProvider>
          {children}
          <Toaster closeButton richColors position="top-right" />
        </ToastStateProvider>
      </TooltipProvider>
    </ThemeProvider>
  )
}
