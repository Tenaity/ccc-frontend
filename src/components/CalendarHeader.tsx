import React from "react";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Label } from "@/components/ui/label";
import { Switch } from "@/components/ui/switch";
import {
  Tabs,
  TabsContent,
  TabsList,
  TabsTrigger,
} from "@/components/ui/tabs";
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from "@/components/ui/select";
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuItem,
  DropdownMenuLabel,
  DropdownMenuSeparator,
  DropdownMenuTrigger,
} from "@/components/ui/dropdown-menu";
import { Card, CardContent } from "@/components/ui/card";
import { Tooltip, TooltipContent, TooltipTrigger } from "@/components/ui/tooltip";
import {
  CalendarIcon,
  DatabaseIcon,
  DownloadIcon,
  Settings2,
  ShuffleIcon,
  Sparkles,
} from "lucide-react";

const months = Array.from({ length: 12 }, (_, index) => index + 1);

export default function CalendarHeader({
  year,
  month,
  setYear,
  setMonth,
  loading,
  onGenerate,
  onShuffle,
  onSave,
  onResetSoft,
  onResetHard,
  onExport,
  exporting = false,
  fillHC,
  setFillHC,
  canGenerate = true,
  onOpenFixedOff,
}: {
  year: number;
  month: number;
  setYear: (y: number) => void;
  setMonth: (m: number) => void;
  loading: boolean;
  onGenerate: () => void;
  onShuffle: () => void;
  onSave: () => void;
  onResetSoft: () => void;
  onResetHard: () => void;
  onExport: () => void;
  exporting?: boolean;
  fillHC: boolean;
  setFillHC: (v: boolean) => void;
  canGenerate?: boolean;
  onOpenFixedOff: () => void;
}) {
  const [activeTab, setActiveTab] = React.useState<"preview" | "manage">(
    "preview"
  );

  const yearOptions = React.useMemo(() => {
    const start = year - 2;
    return Array.from({ length: 5 }, (_, index) => start + index);
  }, [year]);

  return (
    <Card className="mb-6 border-border/60 shadow-sm">
      <CardContent className="flex flex-col gap-6 p-6 md:flex-row md:items-start md:justify-between">
        <div className="flex flex-col gap-4 md:w-1/2">
          <div className="flex flex-wrap items-end gap-4">
            <div className="flex flex-col gap-1">
              <Label htmlFor="year-input" className="text-xs uppercase text-muted-foreground">
                Năm
              </Label>
              <Input
                id="year-input"
                type="number"
                inputMode="numeric"
                value={year}
                onChange={(event) => setYear(Number(event.target.value))}
                className="h-10 w-24"
                disabled={loading}
              />
            </div>
            <div className="flex flex-col gap-1">
              <Label htmlFor="month-select" className="text-xs uppercase text-muted-foreground">
                Tháng
              </Label>
              <Select
                value={String(month)}
                onValueChange={(value) => setMonth(Number(value))}
                disabled={loading}
              >
                <SelectTrigger id="month-select" className="h-10 w-28">
                  <SelectValue placeholder="Chọn tháng" />
                </SelectTrigger>
                <SelectContent>
                  {months.map((m) => (
                    <SelectItem key={m} value={String(m)}>
                      Tháng {m}
                    </SelectItem>
                  ))}
                </SelectContent>
              </Select>
            </div>
            <div className="flex flex-col gap-1">
              <Label htmlFor="year-select" className="text-xs uppercase text-muted-foreground">
                Năm gần đây
              </Label>
              <Select
                value={String(year)}
                onValueChange={(value) => setYear(Number(value))}
                disabled={loading}
              >
                <SelectTrigger id="year-select" className="h-10 w-32">
                  <SelectValue placeholder="Chọn năm" />
                </SelectTrigger>
                <SelectContent>
                  {yearOptions.map((option) => (
                    <SelectItem key={option} value={String(option)}>
                      {option}
                    </SelectItem>
                  ))}
                </SelectContent>
              </Select>
            </div>
          </div>

          <div className="flex flex-wrap items-center gap-3">
            <Tooltip>
              <TooltipTrigger asChild>
                <div className="flex items-center gap-3 rounded-full border border-border/70 bg-muted/40 px-4 py-2">
                  <Switch
                    id="fill-hc-switch"
                    checked={fillHC}
                    onCheckedChange={setFillHC}
                    disabled={loading}
                  />
                  <Label
                    htmlFor="fill-hc-switch"
                    className="text-sm font-medium text-foreground"
                  >
                    Tự động bù HC
                  </Label>
                </div>
              </TooltipTrigger>
              <TooltipContent>
                <p>Bù ca HC khi Generate/Shuffle để tránh thiếu nhân sự.</p>
              </TooltipContent>
            </Tooltip>
            <div className="rounded-full border border-dashed border-border/60 px-4 py-1 text-xs text-muted-foreground">
              <span className="font-medium">Lưu ý:</span> Generate/Shuffle chỉ preview.
            </div>
          </div>
        </div>

        <div className="flex flex-1 flex-col gap-4 md:items-end">
          <Tabs
            value={activeTab}
            onValueChange={(value) => setActiveTab(value as "preview" | "manage")}
            className="w-full md:w-auto"
          >
            <TabsList className="grid w-full grid-cols-2 bg-muted">
              <TabsTrigger value="preview" className="gap-2">
                <Sparkles className="h-4 w-4" /> Preview
              </TabsTrigger>
              <TabsTrigger value="manage" className="gap-2">
                <DatabaseIcon className="h-4 w-4" /> Quản lý
              </TabsTrigger>
            </TabsList>
            <TabsContent
              value="preview"
              className="mt-4 flex flex-col gap-2 md:flex-row"
            >
              <Button
                onClick={onGenerate}
                disabled={loading || !canGenerate}
                className="min-w-[170px]"
              >
                <Sparkles className="mr-2 h-4 w-4" />
                {loading ? "Đang tạo..." : "Generate preview"}
              </Button>
              <Button
                variant="outline"
                onClick={onShuffle}
                disabled={loading || !canGenerate}
                className="min-w-[150px]"
              >
                <ShuffleIcon className="mr-2 h-4 w-4" /> Shuffle preview
              </Button>
            </TabsContent>
            <TabsContent
              value="manage"
              className="mt-4 flex flex-col gap-2 md:flex-row"
            >
              <Button
                variant="secondary"
                onClick={onSave}
                disabled={loading}
                className="min-w-[140px]"
              >
                <DatabaseIcon className="mr-2 h-4 w-4" />
                {loading ? "Đang lưu..." : "Lưu lịch"}
              </Button>
              <DropdownMenu>
                <DropdownMenuTrigger asChild>
                  <Button
                    type="button"
                    variant="outline"
                    disabled={loading}
                    className="min-w-[160px]"
                  >
                    <Settings2 className="mr-2 h-4 w-4" /> Reset
                  </Button>
                </DropdownMenuTrigger>
                <DropdownMenuContent align="end" className="w-56">
                  <DropdownMenuLabel>Reset lịch</DropdownMenuLabel>
                  <DropdownMenuSeparator />
                  <DropdownMenuItem onSelect={onResetSoft}>
                    Reset lịch (soft)
                  </DropdownMenuItem>
                  <DropdownMenuItem onSelect={onResetHard}>
                    Reset DB (hard)
                  </DropdownMenuItem>
                </DropdownMenuContent>
              </DropdownMenu>
            </TabsContent>
          </Tabs>

          <div className="flex flex-wrap justify-end gap-2">
            <Button
              variant="outline"
              onClick={onExport}
              disabled={loading || exporting}
              className="min-w-[140px]"
            >
              <DownloadIcon className="mr-2 h-4 w-4" />
              {exporting ? "Đang xuất..." : "Export CSV"}
            </Button>
            <Button onClick={onOpenFixedOff} disabled={loading} className="min-w-[180px]">
              <CalendarIcon className="mr-2 h-4 w-4" /> Fixed/Off/Holiday
            </Button>
          </div>
        </div>
      </CardContent>
    </Card>
  );
}
