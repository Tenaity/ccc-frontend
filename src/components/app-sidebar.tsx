import * as React from "react"
import { ArrowUpCircleIcon, HelpCircleIcon, SettingsIcon } from "lucide-react"
import { NavLink, useMatch } from "react-router-dom"

import { appRoutes, type AppRoute } from "@/app/routes"
import { NavSecondary } from "@/components/nav-secondary"
import { NavUser } from "@/components/nav-user"
import {
  Sidebar,
  SidebarContent,
  SidebarFooter,
  SidebarGroup,
  SidebarGroupContent,
  SidebarGroupLabel,
  SidebarHeader,
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
} from "@/components/ui/sidebar"
import { ScrollArea, ScrollBar } from "@/components/ui/scroll-area"

const secondaryNav = [
  {
    title: "Settings",
    url: "#settings",
    icon: SettingsIcon,
  },
  {
    title: "Support",
    url: "#support",
    icon: HelpCircleIcon,
  },
]

const currentUser = {
  name: "Customer Care Ops",
  email: "ops@ccc.local",
  avatar: "/avatars/shadcn.jpg",
}

function SidebarNavItem({ route }: { route: AppRoute }) {
  const match = useMatch({
    path: route.path === "/" ? "/" : `${route.path}/*`,
    end: route.path === "/",
  })

  return (
    <SidebarMenuItem key={route.path}>
      <SidebarMenuButton asChild tooltip={route.label} isActive={!!match}>
        <NavLink
          to={route.path}
          end={route.path === "/"}
          className="flex w-full min-w-0 items-center gap-2"
        >
          <route.icon className="size-4 shrink-0" />
          <span className="truncate">{route.label}</span>
        </NavLink>
      </SidebarMenuButton>
    </SidebarMenuItem>
  )
}

export function AppSidebar(props: React.ComponentProps<typeof Sidebar>) {
  return (
    <Sidebar collapsible="offcanvas" {...props}>
      <SidebarHeader>
        <SidebarMenu>
          <SidebarMenuItem>
            <SidebarMenuButton
              asChild
              className="data-[slot=sidebar-menu-button]:!p-1.5"
            >
              <a href="#brand">
                <ArrowUpCircleIcon className="size-5" />
                <span className="text-base font-semibold">
                  Customer Care
                </span>
              </a>
            </SidebarMenuButton>
          </SidebarMenuItem>
        </SidebarMenu>
      </SidebarHeader>
      <SidebarContent>
        <SidebarGroup>
          <SidebarGroupLabel>Navigation</SidebarGroupLabel>
          <SidebarGroupContent>
            <ScrollArea className="h-[calc(100vh-160px)] pr-1">
              <SidebarMenu>
                {appRoutes.map((route) => (
                  <SidebarNavItem key={route.path} route={route} />
                ))}
              </SidebarMenu>
              <ScrollBar orientation="vertical" />
            </ScrollArea>
          </SidebarGroupContent>
        </SidebarGroup>
        <NavSecondary items={secondaryNav} className="mt-auto" />
      </SidebarContent>
      <SidebarFooter>
        <NavUser user={currentUser} />
      </SidebarFooter>
    </Sidebar>
  )
}
