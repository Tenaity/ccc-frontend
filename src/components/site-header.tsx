import { ReactNode } from "react"
import { Link } from "react-router-dom"

import {
  Breadcrumb,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbList,
  BreadcrumbPage,
  BreadcrumbSeparator,
} from "@/components/ui/breadcrumb"
import { Separator } from "@/components/ui/separator"
import { SidebarTrigger } from "@/components/ui/sidebar"

export type SiteHeaderBreadcrumb = {
  label: string
  href?: string
}

export function SiteHeader({
  title,
  description,
  actions,
  breadcrumbs,
}: {
  title: string
  description?: string
  actions?: ReactNode
  breadcrumbs?: SiteHeaderBreadcrumb[]
}) {
  return (
    <header className="group-has-data-[collapsible=icon]/sidebar-wrapper:h-12 sticky top-0 z-10 border-b bg-background/80 backdrop-blur supports-[backdrop-filter]:bg-background/60 transition-[width,height] ease-linear">
      <div className="mx-auto w-full max-w-[1400px] px-3 md:px-6">
        <div className="flex flex-col gap-3 py-3 md:flex-row md:items-center md:justify-between">
          <div className="flex w-full items-start gap-3 md:items-center">
            <div className="flex items-center gap-2">
              <SidebarTrigger className="-ml-1" />
              <Separator orientation="vertical" className="mx-2 hidden data-[orientation=vertical]:h-5 sm:block" />
            </div>
            <div className="flex min-w-0 flex-1 flex-col gap-1">
              <h1 className="text-base font-semibold leading-tight sm:text-lg">
                {title}
              </h1>
              {description ? (
                <p className="text-sm text-muted-foreground line-clamp-2 md:line-clamp-none">
                  {description}
                </p>
              ) : null}
              {breadcrumbs && breadcrumbs.length > 0 ? (
                <Breadcrumb>
                  <BreadcrumbList>
                    {breadcrumbs.map((item, index) => {
                      const isLast = index === breadcrumbs.length - 1

                      return (
                        <BreadcrumbItem key={`${item.label}-${index}`}>
                          {isLast ? (
                            <BreadcrumbPage>{item.label}</BreadcrumbPage>
                          ) : item.href ? (
                            <BreadcrumbLink asChild>
                              <Link to={item.href}>{item.label}</Link>
                            </BreadcrumbLink>
                          ) : (
                            <BreadcrumbPage>{item.label}</BreadcrumbPage>
                          )}
                          {!isLast ? <BreadcrumbSeparator /> : null}
                        </BreadcrumbItem>
                      )
                    })}
                  </BreadcrumbList>
                </Breadcrumb>
              ) : null}
            </div>
          </div>
          {actions ? (
            <div className="flex w-full justify-end gap-2 md:w-auto md:justify-end">
              {actions}
            </div>
          ) : null}
        </div>
      </div>
    </header>
  )
}
