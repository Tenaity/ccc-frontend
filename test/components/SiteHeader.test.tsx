import React from "react"
import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"

import { SiteHeader } from "@/components/site-header"
import { SidebarProvider } from "@/components/ui/sidebar"
import { UiProvider } from "@/components/ui/UiProvider"

function renderSiteHeader(
  ui: React.ReactElement,
  { withRouter = false }: { withRouter?: boolean } = {},
) {
  const content = withRouter ? <MemoryRouter>{ui}</MemoryRouter> : ui

  return render(
    <UiProvider>
      <SidebarProvider defaultOpen>{content}</SidebarProvider>
    </UiProvider>,
  )
}

describe("SiteHeader", () => {
  it("matches mobile snapshot", () => {
    const { asFragment } = renderSiteHeader(
      <SiteHeader
        title="Schedule"
        description="Theo dõi và quản lý phân ca."
        breadcrumbs={[{ label: "Schedule" }]}
      />,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("matches desktop snapshot with actions", () => {
    const { asFragment } = renderSiteHeader(
      <SiteHeader
        title="Schedule"
        description="Theo dõi và quản lý phân ca."
        actions={<button type="button">Tạo mới</button>}
        breadcrumbs={[
          { label: "Dashboard", href: "/" },
          { label: "Schedule" },
        ]}
      />,
      { withRouter: true },
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
