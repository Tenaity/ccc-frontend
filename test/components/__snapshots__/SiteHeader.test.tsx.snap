// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SiteHeader > matches desktop snapshot with actions 1`] = `
<DocumentFragment>
  <div
    class="group/sidebar-wrapper has-data-[variant=inset]:bg-sidebar flex min-h-svh w-full"
    data-slot="sidebar-wrapper"
    style="--sidebar-width: 16rem; --sidebar-width-icon: 3rem;"
  >
    <header
      class="group-has-data-[collapsible=icon]/sidebar-wrapper:h-12 sticky top-0 z-10 border-b bg-background/80 backdrop-blur supports-[backdrop-filter]:bg-background/60 transition-[width,height] ease-linear"
    >
      <div
        class="mx-auto w-full max-w-[1400px] px-3 md:px-6"
      >
        <div
          class="flex flex-col gap-3 py-3 md:flex-row md:items-center md:justify-between"
        >
          <div
            class="flex w-full items-start gap-3 md:items-center"
          >
            <div
              class="flex items-center gap-2"
            >
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground size-7 -ml-1"
                data-sidebar="trigger"
                data-slot="sidebar-trigger"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-panel-left"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="18"
                    rx="2"
                    width="18"
                    x="3"
                    y="3"
                  />
                  <path
                    d="M9 3v18"
                  />
                </svg>
                <span
                  class="sr-only"
                >
                  Toggle Sidebar
                </span>
              </button>
              <div
                class="shrink-0 bg-border/70 h-full w-px mx-2 hidden data-[orientation=vertical]:h-5 sm:block"
                data-orientation="vertical"
                role="none"
              />
            </div>
            <div
              class="flex min-w-0 flex-1 flex-col gap-1"
            >
              <h1
                class="text-base font-semibold leading-tight sm:text-lg"
              >
                Schedule
              </h1>
              <p
                class="text-sm text-muted-foreground line-clamp-2 md:line-clamp-none"
              >
                Theo dõi và quản lý phân ca.
              </p>
              <nav
                aria-label="Breadcrumb"
                class="w-full"
              >
                <ol
                  class="flex flex-wrap items-center gap-1.5 text-sm text-muted-foreground"
                >
                  <li
                    class="inline-flex items-center gap-1"
                  >
                    <a
                      class="inline-flex items-center gap-1 transition-colors hover:text-foreground"
                      data-discover="true"
                      href="/"
                    >
                      Dashboard
                    </a>
                    <span
                      class="text-muted-foreground"
                      role="presentation"
                    >
                      /
                    </span>
                  </li>
                  <li
                    class="inline-flex items-center gap-1"
                  >
                    <span
                      aria-current="page"
                      class="font-medium text-foreground"
                    >
                      Schedule
                    </span>
                  </li>
                </ol>
              </nav>
            </div>
          </div>
          <div
            class="flex w-full justify-end gap-2 md:w-auto md:justify-end"
          >
            <button
              type="button"
            >
              Tạo mới
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>
  <section
    aria-atomic="false"
    aria-label="Notifications alt+T"
    aria-live="polite"
    aria-relevant="additions text"
    tabindex="-1"
  />
</DocumentFragment>
`;

exports[`SiteHeader > matches mobile snapshot 1`] = `
<DocumentFragment>
  <div
    class="group/sidebar-wrapper has-data-[variant=inset]:bg-sidebar flex min-h-svh w-full"
    data-slot="sidebar-wrapper"
    style="--sidebar-width: 16rem; --sidebar-width-icon: 3rem;"
  >
    <header
      class="group-has-data-[collapsible=icon]/sidebar-wrapper:h-12 sticky top-0 z-10 border-b bg-background/80 backdrop-blur supports-[backdrop-filter]:bg-background/60 transition-[width,height] ease-linear"
    >
      <div
        class="mx-auto w-full max-w-[1400px] px-3 md:px-6"
      >
        <div
          class="flex flex-col gap-3 py-3 md:flex-row md:items-center md:justify-between"
        >
          <div
            class="flex w-full items-start gap-3 md:items-center"
          >
            <div
              class="flex items-center gap-2"
            >
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground size-7 -ml-1"
                data-sidebar="trigger"
                data-slot="sidebar-trigger"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-panel-left"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="18"
                    rx="2"
                    width="18"
                    x="3"
                    y="3"
                  />
                  <path
                    d="M9 3v18"
                  />
                </svg>
                <span
                  class="sr-only"
                >
                  Toggle Sidebar
                </span>
              </button>
              <div
                class="shrink-0 bg-border/70 h-full w-px mx-2 hidden data-[orientation=vertical]:h-5 sm:block"
                data-orientation="vertical"
                role="none"
              />
            </div>
            <div
              class="flex min-w-0 flex-1 flex-col gap-1"
            >
              <h1
                class="text-base font-semibold leading-tight sm:text-lg"
              >
                Schedule
              </h1>
              <p
                class="text-sm text-muted-foreground line-clamp-2 md:line-clamp-none"
              >
                Theo dõi và quản lý phân ca.
              </p>
              <nav
                aria-label="Breadcrumb"
                class="w-full"
              >
                <ol
                  class="flex flex-wrap items-center gap-1.5 text-sm text-muted-foreground"
                >
                  <li
                    class="inline-flex items-center gap-1"
                  >
                    <span
                      aria-current="page"
                      class="font-medium text-foreground"
                    >
                      Schedule
                    </span>
                  </li>
                </ol>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
  <section
    aria-atomic="false"
    aria-label="Notifications alt+T"
    aria-live="polite"
    aria-relevant="additions text"
    tabindex="-1"
  />
</DocumentFragment>
`;
